ci/docker/install/ubuntu_base.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# build and install are separated so changes to build don't invalidate
+# the whole docker cache for the image
+
+set -ex
+apt-get update || true
```

Comment:
Curious, why add the boolean OR true condition here ?

Reply:
The "|| true" means that the line will always evaluate with a zero exit code. For scripts with "set -e", any line evaluating with a nonzero exit code will exit the script. I am not sure why we ignore problems on the update, but this is done throughout the CI install scripts.

Reply:
Super! Thanks.

Reply:
Sometimes, some apt repositories are not available and apt-get update fails. If that's the case, we still want to go ahead since all other repositories still updated (think of 1 out of 500 repositories not being available). If the repository we actually need for the installation is actually not available, the install will fail properly and that's fine. But in most cases, it's just an unused and unrelated repository that's not available and thus we just ignore it.